extensions/savvycal/src/listSchedulingLinks.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  ActionPanel,
+  Action,
+  Color,
+  Detail,
+  Icon,
+  List,
+  showHUD,
+  showToast,
+  Toast,
+} from "@raycast/api";
+import { getFavicon } from "@raycast/utils";
+import { useEffect, useState } from "react";
+import { fetchLinksList, toggleLink, SAVVYCAL_BASE_URL } from "./api";
+import { SchedulingLink, PaginatedList } from "./types";
+
+interface Links {
+  links?: SchedulingLink[];
+  error?: unknown;
+}
+
+interface State {
+  state: string;
+}
+
+export const savvycalIcon = getFavicon("https://savvycal.com");
+const greenIcon = { source: Icon.CircleFilled, tintColor: Color.Green };
+const redIcon = { source: Icon.CircleFilled, tintColor: Color.Red };
+
+async function tryToggleLink(link_id: string) {
+  const toast = await showToast({
+    style: Toast.Style.Animated,
+    title: "Toggling the link",
+  });
+
+  try {
+    const { state } = await toggleLink(link_id);
+
+    if (state == "active") {
+      toast.style = Toast.Style.Success;
+      toast.title = "Link is now active!";
+    } else {
+      toast.style = Toast.Style.Success;
+      toast.title = "Link is now inactive.";
+    }
+  } catch (error: unknown) {
+    toast.style = Toast.Style.Failure;
+    toast.title = "Failed to toggle the link";
+  }
+}
+
+export default function listSchedulingLinks() {
+  const [links, setLinks] = useState<Links>({});
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const { entries: entries } = await fetchLinksList();
+        setLinks({ links: entries });
+      } catch (error: unknown) {
+        console.error("Error fetching links:\n", error);
+        setLinks({
+          error: error,
+        });
+        showToast({
+          style: Toast.Style.Failure,
+          title: "Savvycal API Error",
+          message:
+            "Check your Savvycal token + username in the extension preferences",
```

Comment:
There is only a token, right?

Reply:
ah yep, will change this. The initial version asked for a username URL as well

Reply:
Ah, makes sense 🙂